### PR TITLE
Return Option from value function in BST. Closes #276

### LINKED ISSUE #276: Binary Search Tree - value should return an option
[Discussion on Exercism](http://exercism.io/submissions/6e65f3cec5144a9b98180492c11c4e60)

I think it would be nicer if `value` returned an `option` because the empty node has no value. The unit tests currently don't allow this.

### PR DESCRIPTION
**Pull request by ErikSchierboom**
As discussed in #276.

### CODE REVIEW

File: exercises/binary-search-tree/BinarySearchTreeTest.fs [hunk 1]
```diff
@@ -6,40 +6,40 @@ open BinarySearchTree
...
     let tree = singleton 4 |> insert 4
-    Assert.That(tree |> value, Is.EqualTo(4))
-    Assert.That(tree |> left |> Option.map value, Is.EqualTo(Some 4))
+    Assert.That(tree |> value, Is.EqualTo(Some 4))
+    Assert.That(tree |> left |> Option.bind value, Is.EqualTo(Some 4))
 
 [<Test>]
 [<Ignore("Remove to run test")>]
-let ``Inserting greater`` () =
+let ``Inserting greater()`` =
```
Comment (robkeim) [inline exercises/binary-search-tree/BinarySearchTreeTest.fs]: This isn't consistent with the above test cases. I'm not sure the difference between before or after, but I don't see why they wouldn't be consistent.

Comment (ErikSchierboom) [inline exercises/binary-search-tree/BinarySearchTreeTest.fs]: Good find! I'll fix this.

**Comment by ErikSchierboom** [comment]
@robkeim The PR has been updated.

**Comment by robkeim** [comment]
Looks good, thanks @ErikSchierboom! I'm running behind on PRs, I'll try to look through the rest of yours this evening.

**Comment by ErikSchierboom** [comment]
@robkeim No problem! I'm working on the `alphametics` C# exercise now.

**Pull request merged**